### ISSUE TITLE
Ensure Observable.js content works standalone

### ISSUE THREAD

**Issue by mjskay** [journalovi/2023-park-gatherplots#22]
I have made some minor adjustments to the YAML on this paper so that most resources are embedded, to allow the paper to be run offline. This is needed for archiving purposes per #21.

Unfortunately, it looks like the use of observable.js means this paper cannot be rendered fully standalone at this time. I think we'll have to live with that limitation for the time being (observable being an obvious approach for interactivity in articles), and hope that https://github.com/quarto-dev/quarto-cli/issues/1412 is fixed at some point in the future. @cscheid would appreciate any thoughts you have.

**Comment by cscheid** [comment]
Unfortunately that's not something we can easily resolve :(

The full story is the observablehq runtime uses dynamic imports. Dynamic JS imports allow relative path resolution, and as a result, they are not allowed when opening a file in `file://` contexts (understandably, because of security issues). We would have to effectively hard fork the observablehq runtime to not issue those imports and that's something we neither want to nor have the resources to do.

**Comment by mjskay** [comment]
Makes sense re: `file://`. 

I think we'd be okay with things not working with `file://` --- is there a solution (or something on the horizon) that would allow us to make things work if loaded over a local server with no internet connection? e.g. is there something we can change so that we can load modules via the same local server the article is served under (thus, from versions stored in the output folder)?

I think that's a simpler standard for "archivable" than the page being loaded via `file://` since the latter changes the security model, and what we're really worried about is script versions changing or (ultimately) external resources no longer existing at some point in the future. So long as it works via a local server, with all resources served from that local server, we can be a little more confident in an article's longevity...

**Comment by cscheid** [comment]
> is there a solution (or something on the horizon) that would allow us to make things work if loaded over a local server with no internet connection?

I wish, but not really:

- https://github.com/observablehq/stdlib/blob/main/src/require.js#L4
- https://github.com/observablehq/stdlib/blob/main/src/dependencies.js

**Comment by mjskay** [comment]
Ah hrm so if I'm understanding correctly by looking at where `cdn` is used, theoretically we could at least use `setDefaultRequire` to fix uses of `require` but the spots where `cdn` is used directly (eg for `arrow`) would still be broken? Might look for an issue on the observable repos to follow or open one if it doesn't exist.

**Comment by cscheid** [comment]
Yes, that's more or less my understanding. I think ObservableHQ never promised offline support (completely understandably given their platform!)

**Comment by mjskay** [comment]
opened an issue for this here: https://github.com/observablehq/stdlib/issues/394

**Comment by floe** [comment]
I think this may be the right thread/issue to ask, as I'm currently looking into setting up the "stub publication" for this paper on our journal management system.

From what I've read above, there is currently no easy way to turn the interactive paper into a single standalone HTML file, correct? The best we could do would be either a link to the live Github page, or a Zip file of the repo, or ideally both?

**Comment by mjskay** [comment]
For future reference: the issue we're waiting on here is: https://github.com/quarto-dev/quarto/issues/458
